play-validations/memory-footprint/src/main/java/com/google/wear/watchface/dfx/memory/AndroidResourceTable.java

```diff
@@ -0,0 +1,236 @@
+ * <p>
+ * Where obfuscation has been applied, the mapping is derived, for the creation of the
+ * AndroidResource objects. For example, for a logical resource res/raw/watchface.xml, an APK
+ * may in fact store this as res/aB.xml. The resources.arsc file contains this mapping, and the
+ * AndroidResourceTable provides a list of resources with their logical types, names and data.
+ * <p>
+ * Note that more than one AndroidResource object can exist for the given dimensions. For example,
+ * if there is a drawable and a drawable-fr folder, then there may be multiple AndroidResource
+ * entries for drawables with the same type, name and extension. The configuration detail, e.g.
+ * "fr" or "default", is not currently exposed in the AndroidResource objects as it isn't used.
+ */
+public class AndroidResourceTable {
+    // Only certain resource types are of interest to the evaluator, notably, not string resources.
+    private static final Set<String> RESOURCE_TYPES = Set.of("raw", "xml", "drawable", "font", "asset");
+    private static final String RESOURCES_FILE_NAME = "resources.arsc";
+    private final List<AndroidResource> resourceTable;
+
+    private AndroidResourceTable(List<AndroidResource> resources) {
+        this.resourceTable = resources;
+    }
+
+    /**
+     * Creates the table from a path to an AAB structure on the file system.
+     *
+     * @param aabPath The path to the root of the AAB directory.
+     * @return The constructed table.
+     * @throws IOException when the resources file cannot be found, or other IO errors occur.
+     */
+    static AndroidResourceTable createFromAabDirectory(Path aabPath) throws IOException {
+        Stream<Path> childrenFilesStream = java.nio.file.Files.walk(aabPath);
+        ArrayList<AndroidResource> resources = new ArrayList<>();
+        int relativePathOffset = aabPath.toString().length() + 1;
+
+        childrenFilesStream
+            .forEach(
+                filePath -> {
+                    if (AndroidResource.isValidResourcePath(filePath)) {
+                        try {
+                            resources.add(AndroidResource.fromPath(
+                                Paths.get(filePath.toString().substring(relativePathOffset)),
+                                java.nio.file.Files.readAllBytes(filePath)
+                            ));
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
+                        }
+                    } else if (filePath.endsWith("manifest/AndroidManifest.xml")) {
+                        try {
+                            resources.add(new AndroidResource(
+                                "xml",
+                                "AndroidManifest",
+                                "xml",
+                                Paths.get(filePath.toString().substring(relativePathOffset)),
+                                java.nio.file.Files.readAllBytes(filePath)
```

Comment:
I don't know if it's a good idea to store the data for all the resources in memory. I think in the current version, we are returning a `Stream<PackageFile>` and the package files hold all the data in a byte[], but the stream ensures that we're not storing all the PackageFiles in memory at once. We should keep on using streams here.

Reply:
Moved to streamed approach, thanks!